### PR TITLE
add sequtils.delete(index) as re-export of system.delete(index)

### PR DESCRIPTION
**Pull request by timotheecour**
* as discussed earlier
* also renamed the badly named vmutils (which also clashed with an unrelated module)

## future work
- [ ] improve how docgen treats re-export, allowing `export foo {.doc.}` to show the docgen'd API in the re-exporting module; this is a common use case
- [ ] find a way to deprecate system.delete(index) re-export; currently this is not possibly but at least users can now use sequtils.delete(index) (no ambiguity errors since it's the same symbol)

### CODE REVIEW

File: lib/std/private/system_impl.nim [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+when defined(nimSeqsV2):
+  # sync with system.movingCopy
+  template movingCopy(a, b) =
+    a = move(b)
+else:
+  template movingCopy(a, b) =
+    shallowCopy(a, b)
+
+func delete*[T](x: var seq[T], i: Natural) =
+  ## Deletes the item at index `i` by moving all `x[i+1..^1]` items by one position.
```
Comment (konsumlamm) [inline lib/std/private/system_impl.nim:28]:
```suggestion
  ## Deletes the item at index `i` by moving all items in `x[i+1..^1]` by one position.
```

File: lib/std/private/system_impl.nim [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  ##
+  ## This is an `O(n)` operation.
+  ##
+  ## See also:
+  ## * `del <#del,seq[T],Natural>`_ for O(1) operation
+  ##
```
Comment (konsumlamm) [inline lib/std/private/system_impl.nim:34]:
```suggestion
  ## * `del <#del,seq[T],Natural>`_ for a variant that runs in O(1)
```

File: lib/std/private/system_impl.nim [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  runnableExamples:
+    var s = @[1, 2, 3, 4, 5]
+    s.delete(2)
+    doAssert s == @[1, 2, 4, 5]
+
+    doAssertRaises(IndexDefect):
+      s.delete(4)
+
+  if i > high(x):
```
Comment (konsumlamm) [inline lib/std/private/system_impl.nim:43]:
```suggestion
  if i >= len(x):
```
I think this is clearer.

Comment (Araq) [reply lib/std/private/system_impl.nim:43]: Fwiw I think they are equally good.

**Comment by Araq** [comment]
I compiled this program 

```nim

var s = @[1, 2, 4]
delete(s, 500)
echo s
```

It produced no deprecation warning and it raises an exception. What am I missing?

**Comment by Araq** [comment]
Aha

>  find a way to deprecate system.delete(index) re-export; currently this is not possibly but at least users can now use sequtils.delete(index) (no ambiguity errors since it's the same symbol)

Yeah well that's the point of the exercise.

**Comment by Araq** [comment]
Alternative is here, https://github.com/nim-lang/Nim/pull/18507

**Pull request closed without merging**